mythril/analysis/modules/integer.py

```diff
@@ -256,45 +282,62 @@ def _handle_return(state: GlobalState) -> None:
             offset, length = get_concrete_int(stack[-1]), get_concrete_int(stack[-2])
         except TypeError:
             return
+
+        """ TODO: Rewrite this
         for element in state.mstate.memory[offset : offset + length]:
             if not isinstance(element, Expression):
                 continue
             for annotation in element.annotations:
                 if isinstance(annotation, OverUnderflowAnnotation):
+                    
                     state.annotate(
                         OverUnderflowStateAnnotation(
                             annotation.overflowing_state,
                             annotation.operator,
                             annotation.constraint,
                         )
                     )
+        """
 
     def _handle_transaction_end(self, state: GlobalState) -> None:
-        for annotation in cast(
+
+        state_annotations = cast(
             List[OverUnderflowStateAnnotation],
-            state.get_annotations(OverUnderflowStateAnnotation),
-        ):
+            list(state.get_annotations(OverUnderflowStateAnnotation)),
+        )
+
+        if len(state_annotations) == 0:
+            return
+
+        annotations = state_annotations[0].overflowing_state_annotations
+
+        for annotation in annotations:
 
             ostate = annotation.overflowing_state
-            address = _get_address_from_state(ostate)
 
-            if annotation.operator == "subtraction" and self._underflow_cache.get(
-                address, False
-            ):
+            if ostate in self._ostates_unsatisfiable:
```

Comment:
cache is being used, but there isn't any check in here. We can add that to check.